hack/build/binary

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env bash
+set -eu
+
+: "${OUTPUT=./build}"
+: "${PACKAGE=./cmd/dockerd}"
+: "${TARGETPLATFORM=}"
+
+: "${CGO_ENABLED=}"
+: "${GO_LINKMODE=static}"
+: "${GO_LDFLAGS=}"
+: "${GO_BUILDMODE=}"
+: "${GO_BUILDTAGS=apparmor seccomp}"
+
+: "${VERSION=}"
+: "${GITCOMMIT=}"
+: "${BUILDTIME=}"
+: "${PLATFORM=}"
+: "${PRODUCT=}"
+: "${DEFAULT_PRODUCT_LICENSE=}"
+: "${PACKAGER_NAME=}"
+
+# only necessary for non-sandboxed invocation. always off in Dockerfile.cross
+export GO111MODULE=off
+
+if [ -z "$VERSION" ]; then
+	VERSION=$(git describe --match 'v[0-9]*' --dirty='.m' --always --tags | sed 's/^v//' 2> /dev/null || echo "unknown-version")
+fi
+if [ -z "$GITCOMMIT" ]; then
+	GITCOMMIT=$(git rev-parse --short HEAD 2> /dev/null || true)
+fi
+if [ -z "$BUILDTIME" ]; then
+	BUILDTIME=$(date -u --date="@${SOURCE_DATE_EPOCH:-$(date +%s)}" +"%Y-%m-%dT%H:%M:%SZ")
+fi
+
+BIN_NAME=$(basename "$PACKAGE")
+BIN_EXT=
+if [ "$(go env GOOS)" = "windows" ]; then
+	BIN_EXT=.exe
+fi
+
+if [ -z "$CGO_ENABLED" ]; then
+	case "$(go env GOOS)/$(go env GOARCH)" in
+		darwin/* | windows/amd64 | linux/amd64 | linux/arm64 | linux/arm | linux/s390x | linux/ppc64le | linux/riscv*)
+			export CGO_ENABLED=1
+			;;
+		*)
+			export CGO_ENABLED=0
+			;;
+	esac
+fi
+
+if [ -x "$(command -v xx-info)" ]; then
```

Comment:
This block should be removed and shouldn't be needed for anything. Also, avoid using `gcc` in general.

Reply:
Ah indeed I missed that part. Now I see why @cpuguy83 was talking about `CC` and `CXX`.